### PR TITLE
Fix hasSideEffects for AnnotationMethodReflection

### PR DESCRIPTION
**Pull request by VincentLanglet**
Related to https://github.com/phpstan/phpstan/discussions/8586#discussioncomment-4558154

### CODE REVIEW

File: src/Reflection/Annotations/AnnotationMethodReflection.php [hunk 1]
```diff
@@ -109,6 +110,13 @@ public function getThrowType(): ?Type
 
 	public function hasSideEffects(): TrinaryLogic
 	{
+		if (
+			strtolower($this->getName()) !== '__construct'
```
Comment (ondrejmirtes) [inline src/Reflection/Annotations/AnnotationMethodReflection.php]: I think you can delete the thing about __construct.

File: src/Reflection/Annotations/AnnotationMethodReflection.php [hunk 2]
```diff
@@ -109,6 +110,13 @@ public function getThrowType(): ?Type
 
 	public function hasSideEffects(): TrinaryLogic
 	{
+		if (
```
Comment (VincentLanglet) [inline src/Reflection/Annotations/AnnotationMethodReflection.php]: I took the logic from https://github.com/phpstan/phpstan-src/blob/97efd4d36b89fe8a5da39bc71abf5a25b44d4d26/src/Reflection/Php/PhpMethodReflection.php#L410-L415

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: src/Reflection/Annotations/AnnotationMethodReflection.php (hunk 1 above)
Comment (VincentLanglet) [reply src/Reflection/Annotations/AnnotationMethodReflection.php]: Removed

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**